spec/support/handler/profileHandlerspec.js

```diff
@@ -0,0 +1,148 @@
+          //const result = await ph.createProfile(userID);
+          //expect(result.getCode()).toBe(201);
+        //});
+      });
+
+    describe("UpdateProfile method", () => {
+        let ph = new ProfileHandler();
+      
+        beforeEach(() => {
+          userID = generateShortUUID();
+          profileID = generateShortUUID();
+        });
+      
+        it("Req body, username, createdAt, age, bmi, height, weight existence check", async () => {
+          ph.postValue = '{"height": 175, "weight": 70, "bmi": 22.86}'; // Incomplete request body
+          var res = await ph.UpdateProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "height": 175, "weight": 70}'; // Missing age and bmi
+          var res = await ph.UpdateProfile(userID);
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86}'; // Missing height and weight
+          var res = await ph.UpdateProfile(userID);
+          expect(res.getCode()).toBe(400);
+        });
+      
+        it("Req body, username, createdAt, age, bmi, height, weight validity check", async () => {
+          ph.postValue = '{"username": null, "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.UpdateProfile(userID); // Null username
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "invalid_date", "age": 30, "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.UpdateProfile(userID); // Invalid date format
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": "invalid_age", "bmi": 22.86, "height": 175, "weight": 70}';
+          var res = await ph.UpdateProfile(userID); // Invalid age format
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": "invalid_bmi", "height": 175, "weight": 70}';
+          var res = await ph.UpdateProfile(userID); // Invalid BMI format
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": "invalid_height", "weight": 70}';
+          var res = await ph.UpdateProfile(userID); // Invalid height format
+          expect(res.getCode()).toBe(400);
+      
+          ph.postValue = '{"username": "john_doe", "createdAt": "2023-01-01", "age": 30, "bmi": 22.86, "height": 175, "weight": "invalid_weight"}';
+          var res = await ph.UpdateProfile(userID); // Invalid weight format
+          expect(res.getCode()).toBe(400);
+        });
+      
+        it("Update profile successfully", async () => {
```

Comment:
This is actually making the call to the DB this shouldnt happen. maybe a mock is needed